### PR TITLE
DISP-3433: Add getJobNotes to sdk

### PR DESCRIPTION
**Pull request by kswin**
Adds getJobNotes function. Devs could call getCollection directly with the right filters to get the job notes, but feels nicer to call a function that knows the right filters. Also adds unit tests where needed.

### CODE REVIEW

File: lib/dispatch.js [hunk 1]
```diff
@@ -120,10 +121,11 @@ export default class Dispatch {
     let fullPath = endpoint;
     if (filter) {
       const queryParams = [];
-      for (const [k, v] of filter) {
```
Comment (kswin) [inline lib/dispatch.js]: fun fact: `for of` only works on "iterable" object. Those are defined as objects with Symbol.Iterator property. Object literals don't fit that category - which is why i had to use the `for in`

File: lib/dispatch.tests.js [hunk 2]
```diff
@@ -200,4 +201,16 @@ describe('client SDK', () => {
       }).catch(done);
     });
   });
+
+  describe('notes', () => {
+    it('getJobNotes: should call endpoint with correct query string', () => {
+      const client = new Dispatch(testClientID, testClientSecret);
+      const scope = nock('https://api.dispatch.me').get(endpoints.ATTACHMENTS + '?filter[file_token_null]=1&filter[entity_type]=Job&filter[entity_id]=123')
+                                                   .reply(200);
```
Comment (aldeed) [inline lib/dispatch.tests.js]: We do two space indent for things like this, so:

```js
const scope = nock('https://api.dispatch.me')
  .get(endpoints.ATTACHMENTS + '?filter[file_token_null]=1&filter[entity_type]=Job&filter[entity_id]=123')
  .reply(200);
```

You should also use backticks for the string concat:

```js
`${endpoints.ATTACHMENTS}?filter[file_token_null]=1&filter[entity_type]=Job&filter[entity_id]=123`
```

File: lib/dispatch.js (hunk 1 above)
Comment (aldeed) [inline lib/dispatch.js]: yeah, another fun fact is that we use airbnb style guide, which forbids using `for in`, so I guess technically we shouldn't. We do diverge from airbnb on some rules, so maybe we should. We definitely have existing code that uses `for in`. The alternative is 

```js
Object.getOwnPropertyNames(filter).forEach(prop => {
  queryParams.push(`filter[${prop}]=${encodeURIComponent(filter[prop])}`);     
});
```

**Comment by aldeed** [comment]
@kswin I left a couple comments, but generally looks good. At some point I want to discuss what this whole package API should look like at a high level so that we have the structure in place and can build it out with consistency. For example, we might want `client.entities.<entity>.create/update/delete/get+custom` pattern, so the one you added for example would then be `client.entities.job.getNotes(123)` or possibly something with chaining like `client.entities.job(123).getNotes()`.

**Comment by jraede** [comment]
Make sure to also change the `ignore` flags for `npm build` so it doesn't also build the tests. I think it's still set to ignore the `_test.js` instead of `.tests.js`. Agree with @aldeed on the nested API, although I think we can get away with not having it for a while. However `client.entities` makes a lot of sense, e.g. `client.entities.attachment({entity_type: 'Job', entity_id: 10, file_token: null})` would be pretty clean.

**Comment by kswin** [comment]
@aldeed @jraede - I really like those suggestions. If you are cool with it, I'll just go ahead and implement client.entities.attachment - we have edit, delete and add notes in the immediate pipeline so it makes sense for me to set up the pattern at this stage.

**Comment by jraede** [comment]
@kswin works for me. Would also be good to abstract away the differentiator between notes (file_token=null) and photos (file_token!=null) while you're at it

File: lib/dispatch.js [hunk 3]
```diff
@@ -1,11 +1,26 @@
...
+              file_token_null: 1,
+              entity_type: 'Job',
+              entity_id: jobId,
+            };
+            return this.getCollection(endpoints.ATTACHMENTS, filters);
+          }
+        };
+      }
+    };
   }
```
Comment (kswin) [inline lib/dispatch.js]: this object is going to get big quickly. I would appreciate feedback on how to break this out cleanly.

Comment (jraede) [inline lib/dispatch.js]: Might make sense to just break it up into files by entity? The functions could just return `method`, `endpoint`, and `filters`/`body` so they don't have to be aware of the client.

**Comment by jraede** [comment]
@kswin LGTM!

**Pull request merged**